webdev1/webdev4.html

```diff
@@ -0,0 +1,179 @@
+//                fill = '#ca0020';
+//            }
+//            
+//            else if ((thirdPartyPct <= 11) && (repPct > 20) ){
+//                fill = '#ca0020';
+//            }
+//            // 12% or more
+//            else {
+//                fill = '#31a354';
+//            }
+
+            // FIXME: remove existing symbolization and create a diverging
+            // symbology based the democrat or republic vote as a percent
+            // of the total vote.
+            // Use Colorbrewer's 5-class Red-Blue colors for this
+            // http://colorbrewer2.org/#type=diverging&scheme=RdBu&n=5
+
+            // return style specification
+//            return {
+//                color: '#636363',
+//                weight: 2,
+//                fillColor: fill,
+//                fillOpacity: 0.7
+//            }
+ //       },
+
+        // do something to each feature
+        // in this case create a popup
+//        onEachFeature: function (feature, layer) {
+//           var thirdPartyPct = calculateDemPct(feature);
+//            // show name of precinct and third party percent in popup
+//            layer.bindPopup(
+//                "<h3>" + feature.properties.PCTNAME + "</h3>" +
+//                "<p>Third Party Vote: " + thirdPartyPct + "%</p>"
+//           );
+//        }
+    }).addTo(map);
+});	
+
+// a function that takes a feature and returns the percent of the vote
+// that doesn't belong to Republican or Democrat (ie. Third Party)
+function calculatePercent(feature, party) {
+//    console.log(feature);
+    var demVote = feature.properties.USPRSDFL;
+    var repVote = feature.properties.USPRSR;
+    var totalVote = feature.properties.USPRSTOTAL;
+    if (party === 'dem') {
+        var demPct = Math.round(demVote/totalVote*100);
+        return demPct;
+    }
+    // third party vote is whatever is left over from dem and rep vote
+//    var thirdPartyVote = (totalVote - demVote - repVote);
+    // var thirdPartyVote = (totalVote - demVote - repVote);
+    // calculate as a percent of the total vote
+//    var thirdPartyPct = Math.round(thirdPartyVote/totalVote*100);
+//    var demPct = Math.round(demVote/totalVote*100);
+//    var repPct = Math.round(repVote/totalVote*100);
+    
+    // return the third party percent for this feature
+//    return thirdPartyPct;
+}
+    
+function calculateDemPct(feature) {
+    
+//    console.log(feature);
+    var demVote = feature.properties.USPRSDFL;
+    var repVote = feature.properties.USPRSR;
+    var totalVote = feature.properties.USPRSTOTAL;
+    // third party vote is whatever is left over from dem and rep vote
+    var thirdPartyVote = (totalVote - demVote - repVote);
+    
+    // calculate as a percent of the total vote
+    var demPct = Math.round(demVote/totalVote*100);
+    var repPct = Math.round(repVote/totalVote*100);
+    if (demPct > repPct) {
+        return 'dem';
+        }
+    
+    else return 'rep';    
+    
+}
```

Comment:
An interesting take on function returns, but that wasn't really what the lab was asking you to do.